libsolidity/ast/Types.cpp

```diff
@@ -443,13 +443,22 @@ BoolResult AddressType::isImplicitlyConvertibleTo(Type const& _other) const
 
 BoolResult AddressType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	if (_convertTo.category() == category())
+	if ((_convertTo.category() == category()) || isImplicitlyConvertibleTo(_convertTo))
```

Comment:
The first is needed because of payable conversion?

Reply:
Yes. The function `implicitlyConvertibleTo` doesn't allow conversions from non-payable to payable.